src/utils/spawnHelper.js

```diff
@@ -24,10 +24,27 @@ const buildSocketPath = () => tmp.dir()
     return path.join(prefix, cleanedPath, 'echo.sock');
   });
 
+/**
+ * Flow type definition:
+ *
+ * type BuildCredentialsCallbackProcess =
+ *   (
+ *     spawnedProcess: ChildProcess,
+ *     callback: (
+ *       output: string,
+ *       callback: (promptType: string) =>
+ *                 (username: string, password: ?string, cancel: boolean) =>
+ *                 void
+ *       ) => void
+ *     ),
+ *     reject: Error => Promise<void>
+ *   ) =>
+ *   (chunk: Buffer) =>
+ *   string;
+ */
```

Comment:
Do you find it helpful to have the fake 'flow type' here? I am not necessarily opposed to it but we don't use flow in nodegit-lfs yet so it seems a little out of place. Maybe a nice comment or doc explaining what happens would go farther here?